### PR TITLE
added 16IRX8H Lenovo Legion 7i Pro (2023)

### PR DESCRIPTION
**Pull request by realsnick**
###### Description of changes
added 16IRX8H Lenovo Legion 7i Pro (2023)
nvidia is disabled as hyprland fails and that is what i use in my system. if there are specific request i can test more. 

Tested on my own flake with referencing my own nixos-hardware fork locally. 

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -76,6 +76,7 @@
       lenovo-legion-16ach6h-hybrid = import ./lenovo/legion/16ach6h/hybrid;
       lenovo-legion-16ach6h-nvidia = import ./lenovo/legion/16ach6h/nvidia;
       lenovo-legion-16ithg6 = import ./lenovo/legion/16ithg6;
+      lenovo-legion-16irx8h = import ./lenovo/legion/16irx8h;
```
Comment (Mic92) [inline flake.nix:79]: Please also add it to the markdown table in README.md where our ci script picks it up from.

Comment (realsnick) [reply flake.nix:79]: Just learned via https://github.com/johnfanv2/LenovoLegionLinux/issues/81 there is a way to load the lenovo kernel module, will add that and commit together with this.

**Comment by Mic92** [comment]
Just undraft when you are done and I will have a look again.

File: flake.nix (hunk 1 above)
Comment (ulrikstrid) [reply flake.nix:79]: It should be as easy as this:
```nix
  boot.extraModulePackages = [ config.boot.kernelPackages.lenovo-legion-module ];
```

Comment (realsnick) [reply flake.nix:79]: @ulrikstrid added the kernel module and fixed the markdown... I'll suggest to add this kernel module to all of Lenovo legions.

File: lenovo/legion/16irx8h/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ../../../common/hidpi.nix
+  ];
+
+  boot.initrd.kernelModules = ["nvidia"];
+  boot.extraModulePackages = [config.boot.kernelPackages.lenovo-legion-module config.boot.kernelPackages.nvidia_x11];
+
+  hardware = {
+    nvidia = {
+      modesetting.enable = lib.mkDefault true;
+      powerManagement.enable = lib.mkDefault true; #
```
Comment (ulrikstrid) [inline lenovo/legion/16irx8h/default.nix:21]: Remove empty comment

**Review by ulrikstrid — CHANGES_REQUESTED** [review]
Just some empty comments, but other than that it looks good I think

File: lenovo/legion/16irx8h/default.nix (hunk 2 above)
Comment (Mic92) [reply lenovo/legion/16irx8h/default.nix:21]:
```suggestion
      powerManagement.enable = lib.mkDefault true;
```

**Pull request merged**